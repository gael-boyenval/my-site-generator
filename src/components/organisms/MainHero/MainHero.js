import React from 'react';
import styled from 'styled-components';

import { BadassTitle, TinyHeading, LeadBodyCopy } from 'Atoms/Typography';
import Grid from 'Layout/Grid';
import { Padding, Margin } from 'Layout/Spaces';
import Container from 'Layout/Container';
import ButtonLink from 'Atoms/ButtonLink';
import { colors } from 'Styles/tokens';

const Strong = styled.strong`
  font-weight: bold;
  color: ${colors.white.light};
`;

const MainHero = () => (
  <Container>
    <Padding vertical="wide">
      <Grid gutter>
        <Grid.Cell width={8 / 12}>
          <Margin bottom="l">
            <BadassTitle>
              Design.<br />Code.<br />Partage.
            </BadassTitle>
          </Margin>
          <Margin bottom="s">
            <TinyHeading>– Bonjour,</TinyHeading>
          </Margin>
          <LeadBodyCopy>
            Je m’appelle Gaël, suis <Strong>UX/UI designer</Strong> et{' '}
            <Strong>développeur front freelance</Strong> avec plus de{' '}
            <Strong>10 ans d’experience</Strong>. Je travaille sur <Strong>Paris</Strong> en régie
            ou à distance pour des startups et des grand comptes. Vous pouvez aussi faire appel à
            moi pour des missions de <Strong>consulting</Strong>.
          </LeadBodyCopy>
          <Margin top="m">
            <ButtonLink>Mon CV</ButtonLink>
            <br />
            <ButtonLink>les services</ButtonLink>
          </Margin>
        </Grid.Cell>
      </Grid>
    </Padding>
  </Container>
);

export default MainHero;
