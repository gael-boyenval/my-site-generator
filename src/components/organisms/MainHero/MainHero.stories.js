/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import MainHero from './MainHero';

storiesOf('04 - Organisms', module).add('MainHero', withInfo()(() => <MainHero />));
