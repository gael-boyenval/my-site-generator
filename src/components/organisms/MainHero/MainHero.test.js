/* eslint-disable no-undef */
import React from 'react';
import MainHero from './MainHero';

describe('Component MainHero', () => {
  it('should be render without crash', () => {
    const component = shallow(<MainHero title="lorem ipsum dolor " />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
