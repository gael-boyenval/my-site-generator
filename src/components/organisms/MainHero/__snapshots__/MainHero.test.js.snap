// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MainHero should be render without crash 1`] = `
<styled.div>
  <styled.div
    all={false}
    bottom={false}
    horizontal={false}
    inline={false}
    left={false}
    right={false}
    top={false}
    vertical="wide"
  >
    <styled.div
      gutter={true}
    >
      <styled.div
        width={0.6666666666666666}
      >
        <styled.div
          all={false}
          bottom="l"
          horizontal={false}
          inline={false}
          left={false}
          right={false}
          top={false}
          vertical={false}
        >
          <styled.h1>
            Design.
            <br />
            Code.
            <br />
            Partage.
          </styled.h1>
        </styled.div>
        <styled.div
          all={false}
          bottom="s"
          horizontal={false}
          inline={false}
          left={false}
          right={false}
          top={false}
          vertical={false}
        >
          <styled.h4>
            – Bonjour,
          </styled.h4>
        </styled.div>
        <styled.p>
          Je m’appelle Gaël, suis 
          <styled.strong>
            UX/UI designer
          </styled.strong>
           et
           
          <styled.strong>
            développeur front freelance
          </styled.strong>
           avec plus de
           
          <styled.strong>
            10 ans d’experience
          </styled.strong>
          . Je travaille sur 
          <styled.strong>
            Paris
          </styled.strong>
           en régie ou à distance pour des startups et des grand comptes. Vous pouvez aussi faire appel à moi pour des missions de 
          <styled.strong>
            consulting
          </styled.strong>
          .
        </styled.p>
        <styled.div
          all={false}
          bottom={false}
          horizontal={false}
          inline={false}
          left={false}
          right={false}
          top="m"
          vertical={false}
        >
          <ButtonLink>
            Mon CV
          </ButtonLink>
          <br />
          <ButtonLink>
            les services
          </ButtonLink>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
